### PR TITLE
Added CCV2 Build Details View

### PR DESCRIPTION
**Pull request by ABHISHEKGAWANDE01**
Signed-off-by: Abhishek Gawande abhishekgawandeofficial@gmail.com

Screenshot 2024-08-05 at 9 17 27 PM
Screenshot 2024-08-05 at 9 11 14 PM
Screenshot 2024-08-05 at 9 11 35 PM

### CODE REVIEW

File: src/com/intellij/idea/plugin/hybris/tools/ccv2/CCv2Service.kt [hunk 1]
```diff
@@ -530,6 +527,54 @@ class CCv2Service(val project: Project, private val coroutineScope: CoroutineSco
...
+    fun fetchBuildWithCode(
+        subscription: CCv2Subscription,
+        buildCode: String,
+        onStartCallback: () -> Unit,
+        onCompleteCallback: (CCv2BuildDto) -> Unit
+    ) {
+        onStartCallback.invoke()
+
+        coroutineScope.launch {
+            withBackgroundProgress(project, "Fetching Build...$buildCode", true) {
```
Comment (mlytvyn) [inline src/com/intellij/idea/plugin/hybris/tools/ccv2/CCv2Service.kt]: use format `Fetching Build - $buildCode...`

File: src/com/intellij/idea/plugin/hybris/tools/ccv2/CCv2Service.kt [hunk 2]
```diff
@@ -530,6 +527,54 @@ class CCv2Service(val project: Project, private val coroutineScope: CoroutineSco
...
+    fun fetchDeploymentsForBuild(
+        subscription: CCv2Subscription,
+        buildCode: String,
+        onStartCallback: () -> Unit,
+        onCompleteCallback: (List<CCv2DeploymentDto>) -> Unit
+    ) {
+        onStartCallback.invoke()
+        var deployments: List<CCv2DeploymentDto>
+        coroutineScope.launch {
+            withBackgroundProgress(project, "Fetching Deployment for build...", true) {
```
Comment (mlytvyn) [inline src/com/intellij/idea/plugin/hybris/tools/ccv2/CCv2Service.kt]: add buildCode, like `Fetching Deployment for build - $buildCode...`

File: src/com/intellij/idea/plugin/hybris/tools/ccv2/CCv2Service.kt [hunk 3]
```diff
@@ -530,6 +527,54 @@ class CCv2Service(val project: Project, private val coroutineScope: CoroutineSco
...
+                reportProgress(1) { progressReporter ->
+                    val ccv2Token = getCCv2Token(subscription)
+                            try {
```
Comment (mlytvyn) [inline src/com/intellij/idea/plugin/hybris/tools/ccv2/CCv2Service.kt]: reformat the code

File: src/com/intellij/idea/plugin/hybris/tools/ccv2/actions/CCv2BuildActions.kt [hunk 4]
```diff
@@ -108,12 +118,49 @@ class CCv2DeleteBuildAction(
...
+                    e.presentation.text = "Fetch Build"
+
+                    onCompleteCallback.invoke(response)
+                }
+            }
+        )
+    }
+
+    override fun update(e: AnActionEvent) {
+        e.presentation.isEnabled = !fetching && ApplicationSettingsComponent.getInstance().state.ccv2Subscriptions.isNotEmpty()
```
Comment (mlytvyn) [inline src/com/intellij/idea/plugin/hybris/tools/ccv2/actions/CCv2BuildActions.kt:166]: could you please add similar to this check?
```
        val isRightPlace = "GoToAction" != e.place
        e.presentation.isEnabled = isRightPlace && isEnabled()
        e.presentation.isVisible = isRightPlace && e.project
            ?.let { CCv2View.getActiveTab(it) == tab }
            ?: false
```

I'll update other similar actions, we do not want to show our actions in the `Actions GoTo` search.

File: src/com/intellij/idea/plugin/hybris/tools/ccv2/CCv2Service.kt (hunk 2 above)
Comment (ABHISHEKGAWANDE01) [reply src/com/intellij/idea/plugin/hybris/tools/ccv2/CCv2Service.kt]: Resolved.

File: src/com/intellij/idea/plugin/hybris/tools/ccv2/CCv2Service.kt (hunk 3 above)
Comment (ABHISHEKGAWANDE01) [reply src/com/intellij/idea/plugin/hybris/tools/ccv2/CCv2Service.kt]: Resolved.

File: src/com/intellij/idea/plugin/hybris/tools/ccv2/actions/CCv2BuildActions.kt (hunk 4 above)
Comment (ABHISHEKGAWANDE01) [reply src/com/intellij/idea/plugin/hybris/tools/ccv2/actions/CCv2BuildActions.kt:166]: Resolved.

File: src/com/intellij/idea/plugin/hybris/tools/ccv2/CCv2Service.kt (hunk 1 above)
Comment (ABHISHEKGAWANDE01) [reply src/com/intellij/idea/plugin/hybris/tools/ccv2/CCv2Service.kt]: Resolved.

**Review by mlytvyn — APPROVED** [review]

**Pull request merged**